src/cursor/abstract_cursor.ts

```diff
@@ -1101,45 +1101,19 @@ configureResourceManagement(AbstractCursor.prototype);
  *
  * All timeout behavior is exactly the same as the wrapped timeout context's.
  */
-export class CursorTimeoutContext extends TimeoutContext {
-  constructor(
-    public timeoutContext: TimeoutContext,
-    public owner: symbol | AbstractCursor
-  ) {
-    super();
-  }
-  override get serverSelectionTimeout(): Timeout | null {
-    return this.timeoutContext.serverSelectionTimeout;
-  }
-  override get connectionCheckoutTimeout(): Timeout | null {
-    return this.timeoutContext.connectionCheckoutTimeout;
-  }
-  override get clearServerSelectionTimeout(): boolean {
-    return this.timeoutContext.clearServerSelectionTimeout;
-  }
-  override get clearConnectionCheckoutTimeout(): boolean {
-    return this.timeoutContext.clearConnectionCheckoutTimeout;
-  }
-  override get timeoutForSocketWrite(): Timeout | null {
-    return this.timeoutContext.timeoutForSocketWrite;
-  }
-  override get timeoutForSocketRead(): Timeout | null {
-    return this.timeoutContext.timeoutForSocketRead;
-  }
-  override csotEnabled(): this is CSOTTimeoutContext {
-    return this.timeoutContext.csotEnabled();
-  }
-  override refresh(): void {
-    return this.timeoutContext.refresh();
-  }
-  override clear(): void {
-    return this.timeoutContext.clear();
-  }
-  override get maxTimeMS(): number | null {
-    return this.timeoutContext.maxTimeMS;
-  }
-
-  override refreshed(): CursorTimeoutContext {
-    return new CursorTimeoutContext(this.timeoutContext.refreshed(), this.owner);
-  }
+export function CursorTimeoutContext(
```

Comment:
By extending the object, we make it unnecessary to override every single abstract method on the TimeoutContext.